src/mlpack/methods/ann/init_rules/random_init.hpp

```diff
@@ -79,6 +91,18 @@ class RandomInitialization
       Initialize(W.slice(i), rows, cols);
   }
 
+  /**
+   * Initialize randomly the elements of the specified weight 3rd order tensor.
+   *
+   * @param W Weight matrix to initialize.
+   */
+  template<typename eT>
+  void Initialize(arma::Cube<eT>& W)
```

Comment:
Looks good, but don't forget about other initialization rules. Perhaps it's a good idea to move these changes into a separate PR (it's up to you).

Besides, perhaps we should revert `set_size()` back and point out the side effects in the documentation.

Reply:
I agree. I will open a separate PR for that.

Memory leak is still a problem :( I couldn't hunt down the origin.

Reply:
Perhaps that's not your fault. Try to sync your fork with the upstream (there's no memory leaks in the main repo).